sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsHostMemoryStore.scala

```diff
@@ -125,53 +113,58 @@ class RapidsHostMemoryStore(
       catalog: RapidsBufferCatalog,
       stream: Cuda.Stream): Option[RapidsBufferBase] = {
     val wouldFit = trySpillToMaximumSize(other, catalog, stream)
-    // TODO: this is disabled for now since subsequent work will tie this into
-    //   our host allocator apis.
-    if (false && !wouldFit) {
+    if (!wouldFit) {
```

Comment:
I personally think that is what we want, but if anyone has a different opinion I am all ears. We want to get to the point where we have a hard limit on host memory. The reason we don't want to make the changes piecemeal is to reduce the pain a customer might see when running a job. This change is only going to show up when a customer wants to spill from GPU memory, and it is larger than the spill store size. I think that is fairly rare, so I am willing to take the hit.

Reply:
By default the host store is configured to `1GB + pinnedPoolSize`, and `pinnedPoolSize` is defaulted to 0. I think if we raised it to 2GB I'd agree, otherwise it seems fairly common to bump up `batchSizeBytes` to 2GB and those would spill to disk.

Reply:
That said, with such a small spill store, we are bound to spill to disk very often anyway... so I don't know if we are saving too much.